tools/pub_get_offline.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# This python script uses `pub get --offline` to fill in
+# .dart_tool/package_config.json files for Dart packages in the tree whose
+# dependencies should be entirely resolved without requesting data from pub.dev.
+# This allows us to be certain that the Dart code we are pulling for these
+# packages is explicitly fetched by `gclient sync` rather than implicitly
+# fetched by pub version solving, and pub fetching transitive dependencies.
+
+import json
+import os
+import subprocess
+import sys
+
+ALL_PACKAGES = [
+  # 'src/flutter/flutter_frontend_server',
+  os.path.join("src", "flutter", "flutter_frontend_server"),
+  # "src/flutter/tools/const_finder",
+  os.path.join("src", "flutter", "tools", "const_finder"),
+]
+
+
+def FetchPackage(pub, package):
+  try:
+    subprocess.check_output(pub, cwd=package, stderr=subprocess.STDOUT)
+  except subprocess.CalledProcessError as err:
+    print("'%s' failed in '%s' with:\n%s" % (' '.join(cmd), cwd, err.output))
+    return 1
```

Comment:
consider returning the `err. returncode`, or at least printing it here.

Reply:
Alternatively, since you end up just throwing if this returns non-zero on the first one, I think you'd be fine just letting check_output throw, which should print this information.

Reply:
Done.